strutil/map_test.go

```diff
@@ -0,0 +1,80 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil_test
+
+import (
+	"gopkg.in/yaml.v2"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/strutil"
+)
+
+type orderedMapSuite struct{}
+
+var _ = Suite(&orderedMapSuite{})
+
+func (s *orderedMapSuite) TestBasic(c *C) {
+	om := strutil.NewOrderedMap(
+		"K1", "bar",
+		"K2", "baz",
+		"K0", "foo",
+	)
+
+	for i := 0; i < 100; i++ {
+		c.Assert(om.Keys(), DeepEquals, []string{"K1", "K2", "K0"})
+	}
+
+	c.Check(om.Get("K1"), Equals, "bar")
+	c.Check(om.Get("K2"), Equals, "baz")
+	c.Check(om.Get("K0"), Equals, "foo")
+
+	om.Del("K2")
+	c.Assert(om.Keys(), DeepEquals, []string{"K1", "K0"})
+
+	om.Set("K9", "foobar")
+	c.Assert(om.Keys(), DeepEquals, []string{"K1", "K0", "K9"})
+	c.Check(om.Get("K9"), Equals, "foobar")
```

Comment:
Can we please get an extra check ensuring we can replace a key in the middle of the keys and still get the proper semantics on keys and retrieving the modified value?  So far we don't have any tests ensuring a Set with changes works well, I think.